src/main/kotlin/com/dp/advancedgunnerycontrol/weaponais/SpecificAIPluginBase.kt

```diff
@@ -83,43 +80,41 @@ abstract class SpecificAIPluginBase(
     protected fun advanceBaseAI(p0: Float): Boolean {
         if (Settings.forceCustomAI() && isBaseAIOverwritable()) return false
         baseAI.advance(p0)
-        if (isBaseAITargetValid(baseAI.targetShip, baseAI.targetMissile)) {
-            baseAI.targetMissile?.let { targetEntity = it } ?: baseAI.targetShip?.let { targetEntity = it }
-            baseAI.target?.let { targetPoint = it }
+        val ship = baseAI.targetShip
```

Comment:
Here is the only behavior change introduced by the refactor. 

Previously it was possible to end with `targetEntity` not null but `targetPoint` null. At least according to the code. I don't know if it Starsector API caused this.

The current code ensures both `targetEntity` and `targetPoint` stored in `solution` field are not null. If any of the two values is provided null, the entire `solution` field is null.

Reply:
I believe that the baseAI might not provide a non-null target point for do-not-aim-weapons, but I'm not sure.
In which case there'd be a valid firing solution where target == null.
Anyhow, if the base AI for some reason decides to select a target entity but not a target point (and isBaseAITargetValid returns true), I don't see a reason not to trust the baseAI. Hopefully it knows what it's doing xD
So I'd suggest either making aimPoint in FiringSolution nullable, or setting it to a dummy value (e.g. NaN) in that case?
Sounds easier than testing a wide array of weapons to see if the baseAI ever has a target entity but not a target point :P

Reply:
True, I didn't think of that case. I'll update the code.

Reply:
Sorry for the long wait.

I fixed the issue by calling `computePointToAimAt` in case `baseAI.target` is null. That way tags will always have sensible value to work with.